server/src/main/java/io/crate/analyze/AlterTableRenameColumnAnalyzer.java

```diff
@@ -0,0 +1,97 @@
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.analyze;
+
+import java.util.Locale;
+
+import io.crate.analyze.expressions.ExpressionAnalysisContext;
+import io.crate.analyze.expressions.ExpressionAnalyzer;
+import io.crate.analyze.relations.DocTableRelation;
+import io.crate.analyze.relations.NameFieldProvider;
+import io.crate.exceptions.ColumnUnknownException;
+import io.crate.metadata.ColumnIdent;
+import io.crate.metadata.CoordinatorTxnCtx;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.Reference;
+import io.crate.metadata.Schemas;
+import io.crate.metadata.doc.DocTableInfo;
+import io.crate.metadata.table.Operation;
+import io.crate.sql.parser.SqlParser;
+import io.crate.sql.tree.AlterTableRenameColumn;
+import io.crate.sql.tree.ColumnPolicy;
+import io.crate.sql.tree.Expression;
+
+public class AlterTableRenameColumnAnalyzer {
+    private final Schemas schemas;
+    private final NodeContext nodeCtx;
+
+    public AlterTableRenameColumnAnalyzer(Schemas schemas, NodeContext nodeCtx) {
+        this.schemas = schemas;
+        this.nodeCtx = nodeCtx;
+    }
+
+    public AnalyzedAlterTableRenameColumn analyze(AlterTableRenameColumn<Expression> renameColumn,
+                                                  ParamTypeHints paramTypeHints,
+                                                  CoordinatorTxnCtx txnCtx) {
+        if (!renameColumn.table().partitionProperties().isEmpty()) {
+            throw new UnsupportedOperationException("Dropping a column from a single partition is not supported");
+        }
+
+        DocTableInfo tableInfo = (DocTableInfo) schemas.resolveTableInfo(
+            renameColumn.table().getName(),
+            Operation.ALTER,
+            txnCtx.sessionSettings().sessionUser(),
+            txnCtx.sessionSettings().searchPath());
+        var expressionAnalyzer = new ExpressionAnalyzer(
+            txnCtx,
+            nodeCtx,
+            paramTypeHints,
+            new NameFieldProvider(new DocTableRelation(tableInfo)),
```

Comment:
Please use a different (own) field provider instance so avoid using exceptions for the control flow. See e.g.  `TableElementsAnalyzer.resolveField` impl. as an example.

Reply:
Thank you, please check the followup commits.